src/app/dataset/depublication/depublication.component.ts

```diff
@@ -109,24 +109,22 @@ export class DepublicationComponent implements OnDestroy {
   validateRecordIds(control: FormControl): { [key: string]: boolean } | null {
     const val = control.value || '';
     let invalid = false;
-    let misref = false;
-
-    const reg = /(http(s)?:\/\/)?([a-zA-Z]*\/)?([0-9]*\/)?[a-zA-Z\d_]*$/;
-    const reg2 = new RegExp('\\b' + this._datasetId + '\\b');
-
+    const reg = new RegExp(
+      '(^(http(s)?://[^\\s]*)?(^(\\/){1})?([^\\s\\/]+\\/)*(' +
+        this.datasetId +
+        '\\/)+)?[^\\s\\/:\\.]+'
+    );
```

Comment:
I found some obvious problems. I don't think we can aim for perfection here, but perhaps we can at least catch these fairly dramatic cases (assume datasetId = "3"):

'https://path/43/asd' (or any dataset ID ending with 3)
'https:/a/3/c'
'htps:/a/3/c'
'a-_$&#%@)}[*!@#'
'https:///////path//////3/asd'
'https:///path/3/asd'

I tweaked your regular expression a bit, and came up with this:

```
    var reg = new RegExp('^(((http(s)?:\\/\\/)|\\/)?([^\\s\\/:]+\\/)*(' +
        this.datasetId +
        '\\/)+)?[\\w]+$'
    );
```

This seems to prevent all these cases while still allowing the good ones I tried.